File: program/program/convert-string-to-pascalcase/convert_string_to_pascalcase.go
package main

import (
	"fmt"
	"strings"
)

func toPascalCase(str string) string {
	words := strings.Split(str, " ")
	newWords := []string{}
	for _, word := range words {
		firstLetter := word[0]
		restOfWord := word[1:]
		newWord := strings.ToUpper(string(firstLetter)) + restOfWord
		newWords = append(newWords, newWord)
	}
	return strings.Join(newWords, "")
}

func main() {
	str := "hello world"
	fmt.Println("Input string:", str)
	fmt.Println("PascalCase string:", toPascalCase(str))
}

File: program/program/convert-string-to-pascalcase/convert_string_to_pascalcase.r
convertToPascalCase <- function(str) {
  words <- unlist(strsplit(str, " "))  # Split the string into words
  words <- sapply(words, function(word) {
    if (nchar(word) > 0) {
      paste0(toupper(substr(word, 1, 1)), tolower(substr(word, 2, nchar(word))))
    } else {
      ""
    }
  })
  return(paste0(words, collapse = ""))
}

# Test the function with the input "hello world"
input_str <- "hello world"
output_str <- convertToPascalCase(input_str)
print(output_str)
